src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveCMarshal.NativeAot.cs

```diff
@@ -11,6 +11,10 @@ namespace System.Runtime.InteropServices.ObjectiveC
     public static unsafe partial class ObjectiveCMarshal
     {
         private static readonly IntPtr[] s_ObjcMessageSendFunctions = new IntPtr[(int)MessageSendFunction.MsgSendSuperStret + 1];
+        private static bool s_initialized;
+        private static readonly ConditionalWeakTable<object, ObjcTrackingInformation> s_objects = new();
+        private static delegate* unmanaged<IntPtr, int> s_IsTrackedReferenceCallback;
```

Comment:
```suggestion
        private static delegate* unmanaged[SuppressGCTransition]<IntPtr, int> s_IsTrackedReferenceCallback;
```
These callbacks are called in cooperative mode. Does it make sense to capture it in their signature?

Reply:
I combined this suggestion and @AaronRobinsonMSFT's suggestion to use `void*` instead of `IntPtr`. These function pointers now differ from the public [`ObjectiveCMarshal.Initialize` API](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.objectivec.objectivecmarshal.initialize?view=net-7.0). These function pointers are only called from C++, so the difference should have no functional effect.